python_modules/dagster/dagster/_utils/source_position.py

```diff
@@ -87,12 +179,15 @@ def source_position(self) -> SourcePosition:
         """Returns the underlying source position of the object, including
         the source file and line number.
         """
-        return check.not_none(check.not_none(self._source_position_and_key_path).source_position)
+        assert self._source_position_and_key_path
```

Comment:
The check stacking stuff was ugly sure, but generally we have stayed away from using asserts in runtime code so the minor eyesore doesn't feel like it warrants breaking that

Reply:
ah this was an accident from copy pasting the two copies of this file back and forth - will update to dagster-shared now that that is landed